.travis.yml

```diff
@@ -3,32 +3,113 @@
 # This source code is licensed under the Apache License, Version 2.0 found in
 # the LICENSE.txt file in the root directory of this source tree.
 
-language: python
-python:
-  - "3.6"
-os: linux
-dist: trusty
-
-before_install:
-  - sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
-  - sudo apt-get -y update
-  - sudo apt-get -y install g++-7
-  - sudo apt-get -y install cmake
-  - sudo apt-get -y install libopenblas-dev
-
-install:
-  # Installing qiskit-terra master branch...
-  - git clone https://github.com/Qiskit/qiskit-terra.git
-  - cd qiskit-terra
-  - pip install -r requirements.txt
-  - pip install -r requirements-dev.txt
-  - python ./setup.py install
-  - cd ..
-  # Installing qiskit-aer...
-  - pip install -r requirements-dev.txt
-
-script:
-  - python setup.py bdist_wheel -- -DCMAKE_CXX_COMPILER=g++-7 -- -j4
-  - pip install dist/qiskit_aer*whl
-  - mv qiskit qiskit.disable
-  - python -m unittest discover -s test -v
+notifications:
+  email: false
+
+cache: pip
+sudo: false
+
+###############################################################################
+# Anchored and aliased definitions.
+###############################################################################
+
+# These are used for avoiding repeating code, and due to problems with
+# overriding some keys (in particular, "os" and "language: ptyhon") when using
+# the standard travis matrix with stages.
+#
+# This allows re-using different "sets" of configurations in the stages
+# matrix, mimicking a hierarchy:
+# * stage_generic
+#   * stage_linux
+#   * stage_osx
+stage_generic: &stage_generic
+  install:
+    - pip install cython
+    # Installing qiskit-terra master branch...
+    - pip install https://github.com/Qiskit/qiskit-terra/archive/master.zip
+    # Installing qiskit-aer...
+    - pip install -U -r requirements-dev.txt
+
+stage_linux: &stage_linux
+  <<: *stage_generic
+  os: linux
+  dist: xenial
+  language: python
+  python: 3.7
+  sudo: true
+  before_install:
+    - sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
+    - sudo apt-get -y update
+    - sudo apt-get -y install g++-7
+    - sudo apt-get -y install libopenblas-dev
+  script:
+    - python setup.py bdist_wheel -- -DCMAKE_CXX_COMPILER=g++-7 -- -j4
+    - pip install dist/qiskit_aer*whl
+    - python -m unittest discover -s test -v
+
+stage_osx: &stage_osx
+  <<: *stage_generic
+  os: osx
+  osx_image: xcode9.2
```

Comment:
why not use xcode10.2 image since it's the latest version?

Reply:
This is what Terra has, we should sync about this. But moving to 10.2 seems legit